src/views/virtualmachines/tree/VirtualMachineTreeView.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { FC, MouseEvent, useMemo } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom-v5-compat';
+import { VirtualMachineModel } from 'src/views/dashboard-extensions/utils';
+
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { useQueryParamsMethods } from '@kubevirt-utils/components/ListPageFilter/hooks/useQueryParamsMethods';
+import { useIsAdmin } from '@kubevirt-utils/hooks/useIsAdmin';
+import useKubevirtWatchResource from '@kubevirt-utils/hooks/useKubevirtWatchResource';
+import useProjects from '@kubevirt-utils/hooks/useProjects';
+import { VirtualMachineModelGroupVersionKind } from '@kubevirt-utils/models';
+import { convertResourceArrayToMap, getResourceUrl } from '@kubevirt-utils/resources/shared';
+import { FilterValue, useActiveNamespace } from '@openshift-console/dynamic-plugin-sdk';
+import { Grid, GridItem, TreeView, TreeViewDataItem } from '@patternfly/react-core';
+import { OBJECTS_FETCHING_LIMIT } from '@virtualmachines/utils';
+
+import { useSyncClicksEffects } from './hooks/useSyncClicksEffects';
+import {
+  FOLDER_SELECTOR_PREFIX,
+  PROJECT_SELECTOR_PREFIX,
+  VM_FOLDER_LABEL,
+} from './utils/constants';
+import {
+  createTreeViewData,
+  selectedTreeItem,
+  setSelectedTreeItem,
+  treeDataMap,
+} from './utils/utils';
+
+import './VirtualMachineTreeView.scss';
+
+type VirtualMachineTreeViewProps = {
+  onFilterChange?: (type: string, value: FilterValue) => void;
+};
+
+const VirtualMachineTreeView: FC<VirtualMachineTreeViewProps> = ({ children, onFilterChange }) => {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const { setOrRemoveQueryArgument } = useQueryParamsMethods();
+  const pathname = location.pathname;
+  const isAdmin = useIsAdmin();
+  const [activeNamespace, setActiveNamespace] = useActiveNamespace();
+
+  const [vms, vmsLoaded, vmsError] = useKubevirtWatchResource<V1VirtualMachine[]>({
+    groupVersionKind: VirtualMachineModelGroupVersionKind,
+    isList: true,
+    limit: OBJECTS_FETCHING_LIMIT,
+  });
```

Comment:
Does non-admin users will see the threeview?Do we plan to do something different for them here?
This will generate an error as they can't list resources in all namespaces.

Reply:
You're right, I've added a new hook named `useTreeViewData` to get all project, and than for each project we get a watch for the vms.
I'm a bit afraid from this hook since I'm not sure how it will work out at large clusters

Reply:
Yeah right. If the user is admin, we can continue watching vms like this. With just one request. Maybe we can fetch vms by projects only if the user is non admin?